mypy/dmypy.py

```diff
@@ -181,14 +194,19 @@ def start_server(args: argparse.Namespace, allow_sources: bool = False) -> None:
     """Start the server from command arguments and wait for it."""
     # Lazy import so this import doesn't slow down other commands.
     from mypy.dmypy_server import daemonize, Server, process_start_options
-    if daemonize(Server(process_start_options(args.flags, allow_sources),
-                        timeout=args.timeout).serve,
-                 args.log_file) != 0:
-        sys.exit(1)
+    start_options = process_start_options(args.flags, allow_sources)
+    if sys.platform == 'win32':
+        if daemonize(start_options, timeout=args.timeout):
+            sys.exit(1)
+    else:
+        if daemonize(Server(start_options,
+                            timeout=args.timeout).serve,
+                    args.log_file) != 0:
+            sys.exit(1)
     wait_for_server()
 
 
-def wait_for_server(timeout: float = 5.0) -> None:
```

Comment:
Ah, I can probably change this back. My initial implementation had a longer timeout because I was doing something incorrectly which caused things to wait for a while, but things should connect in 5 seconds or less now.